jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -354,82 +354,93 @@ public void scanJars(final WebAppContext context, Collection<Resource> jars, boo
         //Scan jars for META-INF information
         if (jars != null)
         {
-            for (Resource r : jars)
+            try (ResourceFactory.Closeable closeable = ResourceFactory.closeable())
             {
-                if (scanTypes.contains(METAINF_RESOURCES))
-                    scanForResources(context, r, metaInfResourceCache);
-                if (scanTypes.contains(METAINF_FRAGMENTS))
-                    scanForFragment(context, r, metaInfFragmentCache);
-                if (scanTypes.contains(METAINF_TLDS))
-                    scanForTlds(context, r, metaInfTldCache);
+                for (Resource r : jars)
+                {
+                    Resource dir = r;
+
+                    try
+                    {
+                        if (!dir.isDirectory())
+                            dir = closeable.newJarFileResource(dir.getURI());
```

Comment:
Comment to explain that resources may either be a directory, or a jar file that needs to be mounted as a directory.